modules/src/ics02_client/handler/create_client.rs

```diff
@@ -21,33 +19,24 @@ pub fn process(
 ) -> HandlerResult<CreateClientResult, Error> {
     let mut output = HandlerOutput::builder();
 
-    let MsgCreateAnyClient {
-        client_id,
-        client_type,
-        client_state,
-        consensus_state,
-        signer: _,
-    } = msg;
-
-    if ctx.client_state(&client_id).is_some() {
-        return Err(Kind::ClientAlreadyExists(client_id).into());
+    if ctx.client_state(&msg.client_id()).is_some() {
+        return Err(Kind::ClientAlreadyExists(msg.client_id()).into());
     }
 
     output.log("success: no client state found");
 
-    if ctx.client_type(&client_id).is_some() {
-        return Err(Kind::ClientAlreadyExists(client_id).into());
+    if ctx.client_type(&msg.client_id()).is_some() {
```

Comment:
We might consider removing this check, since it's redundant in our implementation. Though it doesn't hurt keeping it.